### ISSUE TITLE
Empty date field is passed through

### ISSUE THREAD

**Issue by moewew** [plk/biber#265]
Via https://tex.stackexchange.com/q/491124/35864

Consider
```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber]{biblatex}

\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@book{appleby,
  author  = {Humphrey Appleby},
  title   = {On the Importance of the Civil Service},
  date    = {},
}
\end{filecontents}

\addbibresource{\jobname.bib}
\addbibresource{biblatex-examples.bib}

\begin{document}
\cite{sigfridsson,appleby}
\printbibliography
\end{document}
```
The `.bbl` produced by Biber (dev and release) for `appleby` has
```
      \field{labeldatesource}{year}
...
      \field{year}{}
```

The empty field value is passed through by the date parsing. This does not happen for other normal fields, there empt field values are simply discarded.

This can cause confusion for `\iffieldundef` tests and indeed the PDF output shows that even the standard styles are ill-equipped to deal with this kind of situation.

So I guess the best course of action would be to just have Biber discard an empty `date` field.

**Comment by plk** [comment]
This was a  bug, fixed in DEV.

**Comment by moewew** [comment]
Works fine now. Thank you.

**Comment by plk** [comment]
Fixed in latest 2.13 release.
